### ISSUE TITLE
Integration with SMT solvers through Schmitty

### ISSUE THREAD

**Issue by jespercockx** [agda/agda2hs#178]
It would be *very* useful if one could use Z3 or another SMT solver to automatically dismiss proof obligations for obvious statements. There is already such an integration with Agda: Schmitty (https://github.com/wenkokke/schmitty/). We should check what is needed to integrate Schmitty into an Agda2Hs project and whether we can do anything to make it easier to use (e.g. provide predefined theories for types and functions from the Prelude).

**Comment by uhbif19** [comment]
Why this would be different from regular Agda? I thought that most of such proofs would be erased anyway.

**Comment by jespercockx** [comment]
Yes, the proofs would indeed be erased. What does need to be done are the issues Wen lists above: the dependencies on stdlib and agdarsec, and the work of mapping the types and functions in the Agda2Hs prelude to the appropriate SMT theories.

**Comment by uhbif19** [comment]
@jespercockx Could not `agda-stdlib` be mapped instead? I understand that is not in the paper, but I do not understand why. Like code base already has some lists for specific types translation, could not some pragma or typeclass be used to define new Agda -> Haskell translations for terms of well-known Agda libs?

**Comment by wenkokke** [comment]
> @jespercockx Could not `agda-stdlib` be mapped instead? I understand that is not in the paper, but I do not understand why. Like code base already has some lists for specific types translation, could not some pragma or typeclass be used to define new Agda -> Haskell translations for terms of well-known Agda libs?

That'd work, but then you still can't use Schmitty with conventional agda2hs, right?
